client/record/cli/sendtx.go

```diff
@@ -45,15 +46,24 @@ func GetCmdSubmitFile(storeName string, cdc *wire.Codec) *cobra.Command {
 			// --onchain-data has a high priority over --file-path
 			if len(onchainData) != 0 {
 				dataSize = int64(binary.Size([]byte(onchainData)))
+				if dataSize >= recordClient.UploadLimitOfOnchain {
+					fmt.Printf("File %s is too large, upload limit is %d bytes.\n", filePath, recordClient.UploadLimitOfOnchain)
+					return err
+				}
 				sum := sha256.Sum256([]byte(onchainData))
-				recordHash = hex.EncodeToString(sum[:])
+				recordHash = hex.EncodeToString(sum[:recordClient.IpfsHashLength/2])
```

Comment:
Why cut the hash ??

Reply:
How can prove that after discards some bytes of the hash, it's still safe to the hash collision ？

Reply:
Because onchaindata hash is caculated from SHA256 which length is longer than ipfs hash length(i.e. 46 charater) . To be consistent, onchaindata is cut. It's on the one hand. At the same time ipfs hash which is 23 bytes long supports unique data index. So I think it's safe to cut aforementioned onchaindata hash. It's on the one hand. On the other hand, indeeed, there's brute force risk, if we narrowed down this hash range. So, currently, for the safety, I will keep integral hash.